### ISSUE TITLE
useLazyQuery typescript error called is always true after upgrading from 3.4.17 -> 3.5.4

### ISSUE THREAD

**Issue by anark** [apollographql/apollo-client#9102]
**Intended outcome:**

`const [query, {called}] = useLazyQuery()` the `called` variable should have typescript `boolean` type

**Actual outcome:**

`called` variable has typescript type `true`

**How to reproduce the issue:**

**Versions**

```
  System:
    OS: macOS 12.0.1
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 3.0.0 - ~/Development/sky-pilot/node_modules/.bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 95.0.4638.69
    Safari: 15.1
  npmPackages:
    @apollo/client: 3.5.4 => 3.5.4 
    @apollo/react-testing: 4.0.0 => 4.0.0 
```

**Comment by anark** [comment]
Still seeing this on 3.5.6.  Any ideas for a fix for this?

**Comment by benjamn** [comment]
The typing of `called` should be fixed in v3.5.8, thanks to #9328.
